src/abort_handler.rs

```diff
@@ -0,0 +1,17 @@
+use libc::{c_int, c_void, sighandler_t, signal, SIGABRT, SIGILL};
+
+extern "C" fn handler(_: c_int) {
+    eprintln!("Abort: Unexpected error.");
+    std::process::exit(127);
+}
+
+fn get_handler() -> sighandler_t {
+    handler as extern "C" fn(c_int) as *mut c_void as sighandler_t
+}
+
+pub fn setup() {
+    unsafe {
+        signal(SIGABRT, get_handler());
+        signal(SIGILL, get_handler());
```

Comment:
Linux man pages seem to prefer `sigaction` to `signal`. Should we still use `signal`?

If we are planning to handle some events in separate threads: "The effects of signal() in a multithreaded process are unspecified."

Reply:
Good catch. `sigaction` isn't available on Windows, but this won't work there anyways.